File: LR2/7/7.py
def count_points(rings):
    # Инициализируем список из 10 стержней, каждый представлен множеством для хранения цветов
    rods = [set() for _ in range(10)]
    n = len(rings)
    
    # Обрабатываем каждую пару символов (цвет и позиция)
    for i in range(0, n, 2):
        # Проверяем, что пара символов существует (не выходит за пределы строки)
        if i + 1 >= n:
            break
        
        color = rings[i]
        pos_char = rings[i + 1]
        
        # Пропускаем недопустимые цвета
        if color not in {'R', 'B', 'W'}:
            continue
        
        # Пропускаем недопустимые позиции (не цифра или не в диапазоне 0-9)
        if not pos_char.isdigit():
            continue
        
        pos = int(pos_char)
        if 0 <= pos <= 9:
            rods[pos].add(color)  # Добавляем цвет в соответствующий стержень
    
    # Считаем стержни с тремя разными цветами
    return sum(1 for rod in rods if len(rod) == 3)

def main():
    input_str = input().strip()
    print(count_points(input_str))

if __name__ == "__main__":
    main()

File: LR2/7/7.kt
// Функция для подсчета стержней с кольцами всех трех цветов
fun countPoints(rings: String): Int {
    // Класс, представляющий стержень с флагами наличия цветов
    class Rod {
        var hasR = false
        var hasB = false
        var hasW = false
    }

    // Инициализируем массив из 10 стержней
    val rods = Array(10) { Rod() }

    // Проходим по входной строке шагом 2 (цвет и позиция)
    for (i in 0 until rings.length step 2) {
        // Проверка на выход за границы строки для пары
        if (i + 1 >= rings.length) break

        val color = rings[i]
        val posChar = rings[i + 1]

        // Проверка, что позиция - это цифра от 0 до 9
        if (!posChar.isDigit()) continue
        val pos = posChar.toString().toInt()
        if (pos !in 0..9) continue

        // Обновляем флаги стержня в зависимости от цвета
        when (color) {
            'R' -> rods[pos].hasR = true
            'B' -> rods[pos].hasB = true
            'W' -> rods[pos].hasW = true
            // Игнорируем недопустимые цвета
            else -> continue
        }
    }

    // Подсчитываем стержни с тремя цветами
    return rods.count { it.hasR && it.hasB && it.hasW }
}

fun main() {
    println("Input:")
    val input = readLine() ?: "" // Чтение входной строки
    println(countPoints(input))
}